pupil_src/shared_modules/video_capture/uvc_backend.py

```diff
@@ -790,9 +793,15 @@ def dev_selection_list():
     def activate(self, source_uid):
         if not source_uid:
             return
-        if not uvc.is_accessible(source_uid):
-            logger.error("The selected camera is already in use or blocked.")
+
+        try:
+            if not uvc.is_accessible(source_uid):
+                logger.error("The selected camera is already in use or blocked.")
+                return
+        except ValueError as ve:
```

Comment:
When and by what is this `ValueError` triggered?

Reply:
For the following scenario:
1. switch from Test Image to UVC backend
2. hit Auto-select
3. (all cameras active now)
4. unplug headset, hit the button again shortly after


Here, `UVCManager.auto_activate_source` calls the `self.activate(source_id)` with a valid argument. But when in `UVCManager.activate`, the `source_uid` is not `None`, but also invalid somehow. 

This raises a `ValueError` in `if not uvc.is_accessible(source_uid)`.

Reply:
I hope I get it right this time. Both `StopIteration` and `for/else` are fairly new to me.

Reply:
Ah, thank you. I guess `uvc.is_accessible()` expects a value that is part of `uvc.Device_List()`